theseus/embodied/robot/joint.py

```diff
@@ -217,3 +217,158 @@ def _rotation_impl(self, angle: torch.Tensor) -> torch.Tensor:
         rot[:, 0, 1] = -rot[:, 1, 0]
         rot[:, 1, 1] = rot[:, 0, 0]
         return rot
+
+
+class _PrismaticJointImpl(Joint):
+    def __init__(
+        self,
+        name: str,
+        parent: int = -1,
+        child: int = -1,
+        origin: Optional[torch.Tensor] = None,
+        dtype: Optional[torch.dtype] = None,
+    ):
+        super().__init__(name, parent, child, origin, dtype)
+
+    def dof(self) -> int:
+        return 1
+
+    @abc.abstractmethod
+    def _translation_impl(self, angle: torch.Tensor) -> torch.Tensor:
```

Comment:
Same comment as in `RevoluteJoint` regarding a `translation()` method, with similar structure.

Reply:
Refactored.